sh/shadowenv.bash.in

```diff
@@ -3,9 +3,13 @@ __shadowenv_hook() {
   if [[ "$1" == "preexec" ]]; then
     flags+=(--silent)
   fi
```

Comment:
I think it would be nicer if we did this the other way around, defining the variable outside of the function, and unsetting it after the first run. Would that work?

Reply:
I liked how keeping it all in the function made it a little more self-contained and wouldn't depend on something being set elsewhere, but it has made it a little awkward to name the variable... is that mainly why you'd think it would be nicer if defined outside?

no strong feelings at the end of the day though

Reply:
The reason I prefer what @burke suggests is that it leaves the environment unpolluted by an environment variable for most of the time.  It's something we strive for in the tools we write.

Reply:
that's fair, i'll update